### PR TITLE
Move and clean some pylab examples

### PR DESCRIPTION
**Pull request by dstansby**
Some more pylab moves (maybe I'll get them all done by 2.1...). A couple of notes:

- I've PEP8 cleaned the moved files
- I've changed a couple of example colours to the default colour cycle

### CODE REVIEW

File: examples/subplots_axes_and_figures/axhspan_demo.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+"""
+============
+Axhspan Demo
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/axhspan_demo.py]: Should be "axhspan" (lowercase).  Probably there are other cases of wrong uppercasing, haven't checked.

File: examples/text_labels_and_annotations/fancyarrow_demo.py [hunk 2]
```diff
@@ -28,15 +28,16 @@ def to_texstring(s):
...
-    x = 3.2 + (i//nrow)*4
+    x = 3.2 + (i // nrow) * 4
     y = (figheight - 0.7 - i % nrow)  # /figheight
     p = mpatches.Circle((x, y), 0.2)
     ax.add_patch(p)
 
     ax.annotate(to_texstring(stylename), (x, y),
                 (x - 1.2, y),
-                #xycoords="figure fraction", textcoords="figure fraction",
+                # xycoords="figure fraction", textcoords="figure fraction",
```
Comment (anntzer) [inline examples/text_labels_and_annotations/fancyarrow_demo.py]: If the comment is not relevant, just remove it.

File: examples/subplots_axes_and_figures/axhspan_demo.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# Draw a thick red hline at y=0 that spans the xrange
+plt.axhline(linewidth=8, color='#d62728')
+
+# Draw a default hline at y=1 that spans the xrange
+plt.axhline(y=1)
+
+# Draw a default vline at x=1 that spans the yrange
+plt.axvline(x=1)
+
+# Draw a thick blue vline at x=0 that spans the upper quadrant of
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/axhspan_demo.py]: this comment and the one below do not need to be linewrapped

File: examples/subplots_axes_and_figures/axes_demo.py [hunk 4]
```diff
@@ -14,9 +14,9 @@
 # create some data to use for the plot
 dt = 0.001
 t = np.arange(0.0, 10.0, dt)
-r = np.exp(-t[:1000]/0.05)               # impulse response
+r = np.exp(-t[:1000] / 0.05)               # impulse response
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/axes_demo.py]: comment should be closer

**Comment by tacaswell** [comment]
This seems to have broken a link

```
URL        `../../gallery/pylab_examples/bar_stacked.html'
Name       `bar_stacked.py'
Parent URL file:///home/travis/build/matplotlib/matplotlib/doc/build/html/tutorials/01_introductory/sample_plots.html, line 237, col 2
Real URL   file:///home/travis/build/matplotlib/matplotlib/doc/build/html/gallery/pylab_examples/bar_stacked.html
Check time 0.003 seconds
Result     Error: URLError: <urlopen error [Errno 2] No such file or directory: '/home/travis/build/matplotlib/matplotlib/doc/build/html/gallery/pylab_examples/bar_stacked.html'>
```

**Comment by QuLogic** [comment]
Hmm, can you rebase this?

**Comment by dstansby** [comment]
Rebased, would be worth someone checking the file that is deleted and re-added in the diff.

**Pull request merged**